### PR TITLE
deprecation: Remove OQC

### PR DESCRIPTION
**Pull request by speller26**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the PR title format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#PR-title-format)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/devices/devices.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class _Xanadu(str, Enum):
     # DWave = _DWave
     IonQ = _IonQ
     IQM = _IQM
-    OQC = _OQC
+    # OQC = _OQC
```
Comment (virajvchaudhari) [inline src/braket/devices/devices.py:61]: Intentionally commenting this instead of removing this from the Enums?

Comment (speller26) [reply src/braket/devices/devices.py:61]: Just doing the same thing we've done with D-Wave and Xanadu

**Review by virajvchaudhari — DISMISSED** [review]

**Review by virajvchaudhari — DISMISSED** [review]

**Comment by virajvchaudhari** [comment]
There are few more [places](https://github.com/search?q=repo%3Aamazon-braket%2Famazon-braket-sdk-python%20OQC&type=code) to remove?

File: test/integ_tests/test_device_creation.py [hunk 2]
```diff
@@ -77,7 +73,7 @@ def test_get_devices_others():
 
 def test_get_devices_all(braket_devices):
     result_arns = [result.arn for result in braket_devices]
-    for arn in [RIGETTI_ARN, IONQ_ARN, SIMULATOR_ARN, OQC_ARN]:
+    for arn in [RIGETTI_ARN, IONQ_ARN, SIMULATOR_ARN]:
```
Comment (virajvchaudhari) [inline test/integ_tests/test_device_creation.py:76]: Why haven't we added IQM_ARN here as well?

Comment (speller26) [reply test/integ_tests/test_device_creation.py:76]: Sure, I'm fine with that

**Review by virajvchaudhari — APPROVED** [review]

**Pull request merged**